src/Query/Query.php

```diff
@@ -109,12 +109,19 @@ public function orderBy(array $order): static
                 continue;
             }
 
-            if (!is_int($value)) {
+            if ($value instanceof Direction) {
+                $this->order[] = new OrderField($key, $value);
+                continue;
+            }
+
+            if (is_string($value)) {
                 $value = match (strtoupper($value)) {
-                    "ASCENDING", "ASC" => OrderField::ASCENDING,
-                    "DESCENDING", "DESC" => OrderField::DESCENDING,
+                    "ASCENDING", "ASC" => Direction::ASCENDING,
+                    "DESCENDING", "DESC" => Direction::DESCENDING,
                     default => throw new InvalidArgumentException('Argument $order contains invalid order direction: ' . $value),
                 };
+            } else {
```

Comment:
Why not just use `} else if (!($value instanceof Direction)) {` and remove the other if branch above?

Reply:
Like this?